src/main/java/com/okta/sdk/framework/FilterBuilder.java

```diff
@@ -204,24 +168,72 @@ public FilterBuilder lessThanOrEqual(DateTime value) {
     }
 
     public FilterBuilder and() {
-        STRING_BUILDER.append(AND_SIGN);
+        filterBuilder.append(AND_SIGN);
+        totalNumberOfLogicalOperators += 1;
+        checkComplexity();
         return this;
     }
 
     public FilterBuilder and(FilterBuilder filterBuilder) {
-        STRING_BUILDER.append(AND_SIGN);
+        this.filterBuilder.append(AND_SIGN);
         return where(filterBuilder);
     }
 
     public FilterBuilder or() {
-        orCount++;
-        STRING_BUILDER.append(OR_SIGN);
+        filterBuilder.append(OR_SIGN);
+        totalNumberOfLogicalOperators += 1;
+        checkComplexity();
         return this;
     }
 
     public FilterBuilder or(FilterBuilder filterBuilder) {
-        orCount++;
-        STRING_BUILDER.append(OR_SIGN);
+        this.filterBuilder.append(OR_SIGN);
         return where(filterBuilder);
     }
+
+    private FilterBuilder equalTo() {
+        filterBuilder.append(EQUAL_SIGN);
+        return this;
+    }
+
+    private FilterBuilder contains() {
+        filterBuilder.append(CONTAIN_SIGN);
+        return this;
+    }
+
+    private FilterBuilder startsWith() {
+        filterBuilder.append(STARTS_WITH_SIGN);
+        return this;
+    }
+
+    private FilterBuilder greaterThan() {
+        filterBuilder.append(GREATER_THAN_SIGN);
+        return this;
+    }
+
+    private FilterBuilder greaterThanOrEqual() {
+        filterBuilder.append(GREATER_THAN_OR_EQUAL_SIGN);
+        return this;
+    }
+
+    private FilterBuilder lessThan() {
+        filterBuilder.append(LESS_THAN_SIGN);
+        return this;
+    }
+
+    private boolean checkComplexity() {
```

Comment:
I thought we decided to let the server enforce this. We never want to end up in a situation where the server loosens up its constraints (allows more operators) and then have the SDK explicitly precluding use of the new server abilities.